src/components/manage/Add/Add.jsx

```diff
@@ -220,6 +233,15 @@ class Add extends Component {
       const blocksLayoutFieldname = getBlocksLayoutFieldname(
         this.props.schema.properties,
       );
+      const translationObject = this.props.location?.state?.translationObject;
+      const translateTo = this.props.location?.state?.language;
+
+      if (translationObject && blocksFieldname && blocksLayoutFieldname) {
+        //copy blocks from translationObject
+        this.initialBlocks = translationObject[blocksFieldname];
+        this.initialBlocksLayout =
+          translationObject[blocksLayoutFieldname].items;
```

Comment:
@giuliaghisini You should regenerate block uids for translation in order to avoid duplicates. 

I would also suggest to keep a reference to the original block uid as we do for `@layout`:

https://github.com/plone/volto/blob/a2ade17a8fdbe5266de845b2dfe1e44c78e95802/src/components/manage/Add/Add.jsx#L263-L264

Suggested key `@canonical`

Also, it would be nice to split this logic within dedicated methods and remove it from render.

Reply:
@giuliaghisini Also, you should move:

https://github.com/plone/volto/blob/a2ade17a8fdbe5266de845b2dfe1e44c78e95802/src/components/manage/Add/Add.jsx#L239-L244

after the [schema logic](https://github.com/plone/volto/blob/a2ade17a8fdbe5266de845b2dfe1e44c78e95802/src/components/manage/Add/Add.jsx#L246-L267) and use `initialBlocks`/`initialBlocksLayout` instead of `this.initialBlocks` / `this.initialBlocksLayout` as if there are schemaBlocks defined the translation will lose your changes.